source/base/utilities.cc

```diff
@@ -219,7 +219,7 @@ namespace Utilities
                (name[0] == ' '))
           name.erase (0,1);
 
-        while (name[name.length()-1] == ' ')
+        while (name[name.length()-1] == ' ' && name.length() != 0)
```

Comment:
You need to turn the condition around if you want to avoid the invalid access: `&&` expressions are evaluated left to right, so you still get the invalid access because it happens first.